e2e/test-tomcat.py

```diff
@@ -25,12 +25,12 @@ def main():
 
     subprocess.check_output(['mvn', '-q', 'tomcat7:help'], cwd='java-goof')
 
-    victim = subprocess.Popen(['mvn', 'tomcat7:run'], env={
-        'MAVEN_OPTS': '-javaagent:{}=file://{}'.format(
-            path.join(os.getcwd(), 'build/libs/snyk-java-runtime-agent.jar'),
-            h.config_path
-        )
-    }, cwd='java-goof')
+    opts_env = os.environ.copy()
```

Comment:
do we know what was the problem with the previous way?
was `env` overridden and lost other vars?

Reply:
It appears that, on some Python versions, `PATH` defaults to empty, and on others, it defaults to some secret value that was recorded at startup.

https://docs.python.org/3/library/subprocess.html documents:

> If env is not None, it must be a mapping that defines the environment variables for the new process; these are used instead of the default behavior of inheriting the current process’ environment.
> Note If specified, env must provide any variables required for the program to execute. On Windows, in order to run a side-by-side assembly the specified env must include a valid SystemRoot.

...which is the behaviour :travis: wants.

The new code works on either setup, so is arguably right.